test/unit/services/users.test.js

```diff
@@ -100,4 +105,49 @@ describe("Users services", function () {
       expect(newUsername).to.equal("john-doe-1");
     });
   });
+
+  describe("fetchUsersWithAbandonedTasks", function () {
+    beforeEach(async function () {
+      await cleanDb();
+
+      const userPromises = abandonedUsersData.map((user) => userModel.add(user));
+      await Promise.all(userPromises);
+
+      const taskPromises = abandonedTasksData.map((task) => tasksModel.add(task));
+      await Promise.all(taskPromises);
+    });
+
+    afterEach(async function () {
+      await cleanDb();
+    });
+
+    it("should fetch users with abandoned tasks", async function () {
+      const usersWithAbandonedTasks = await getUsersWithIncompleteTasks();
+
+      expect(usersWithAbandonedTasks).to.be.an("array");
+      expect(usersWithAbandonedTasks).to.have.lengthOf(2);
+    });
+
+    it("should not include user who are present in discord or not archived", async function () {
+      const usersWithAbandonedTasks = await getUsersWithIncompleteTasks();
+
+      usersWithAbandonedTasks.forEach((user) => {
+        expect(user.roles.in_discord).to.not.equal(true);
+        expect(user.roles.archived).to.not.equal(false);
+      });
+    });
+
+    it("should return an empty array if there are no users with abandoned tasks", async function () {
+      await cleanDb();
+
+      const activeUser = abandonedUsersData[2];
+      await userModel.add(activeUser);
+
+      const activeTask = abandonedTasksData[3];
+      await tasksModel.add(activeTask);
+      const usersWithAbandonedTasks = await getUsersWithIncompleteTasks();
+      expect(usersWithAbandonedTasks).to.be.an("array");
+      expect(usersWithAbandonedTasks).to.have.lengthOf(0);
+    });
```

Comment:
add more asserts

Reply:
The test case checks whether an empty array is returned when there are no users with abandoned tasks, and the existing assertions validate this behaviour. Could you clarify what kind of additional assertions you are expecting?

Reply:
Resolved